libmcount/mcount.c

```diff
@@ -552,6 +587,22 @@ void mcount_entry_filter_record(struct mcount_thread_data *mtdp,
 				struct mcount_regs *regs)
 {
 	mtdp->record_idx++;
+
+	/* script hooking for function entry */
+	if (script_str) {
+		unsigned long entry_addr = rstack->child_ip;
+		struct sym *sym = find_symtabs(&symtabs, entry_addr);
+		char *symname = symbol_getname(sym, entry_addr);
+		struct script_info scinfo = {
+			.tid = gettid(mtdp),
+			.depth = rstack->depth,
+			.timestamp = rstack->start_time,
+			.address = entry_addr,
+			.symname = symname,
+		};
+		script_uftrace_entry(&scinfo);
+		symbol_putname(sym, symname);
+	}
 }
```

Comment:
I'd like to keep this as small and fast as possible.  So I think you don't need to add the script handling here and change `cmd-record.c:can_use_fast_libmcount()` return false when "UFTRACE_SCRIPT" env is set.